cmd/controller-manager/app/server.go

```diff
@@ -114,6 +114,37 @@ func NewControllerManagerCommand() *cobra.Command {
 	return cmd
 }
 
+func Run(s *options.KubeSphereControllerManagerOptions, configCh <-chan controllerconfig.Config, ctx context.Context) error {
+	ictx := controllerconfig.Context{}
```

Comment:
Why do we need to create a new Context{}? Can we just use default context with cancel function?
```
ictx, cancel := context.WithCancel(context.Background)
```

Reply:
The default context (`signals.SetupSignalHandler()`) is to control the entire program life cycle,  
A new context is created here to control the life cycle of the controller-manager(all controllers, sharedInformer, webhook etc.)

Reply:
Yes, I understand the necessity of a new context, my concern is why not use `context.WithCancel` rather than create a new type?

Reply:
I encapsulated the context so that ks-apiserver and controller-manager use the same context API. This is really confusing, I will remove it and add a comment